src/main/java/org/terasology/climateConditions/HyperthermiaComponent.java

```diff
@@ -23,4 +23,41 @@
     public float regenMultiplier = 0.8f;
     public float maxHealthMultiplier = 0.8f;
     public float thirstMultiplier = 2f;
+    public float hyperthermiaLevel;
+    //The higher the value of the modifier, the more dangerous the effects of Hyperthermia.
+    public float allEffectModifier;
+
+    HyperthermiaComponent() {
+        hyperthermiaLevel = 1;
+        allEffectModifier = 1;
+    }
+
+    HyperthermiaComponent(int level) {
+        hyperthermiaLevel = level;
+        allEffectModifier = level;
+    }
+
+    public float getEffectiveWalkSpeedMultiplier() {
+        return walkSpeedMultiplier / allEffectModifier;
+    }
```

Comment:
I'd say these methods need more JavaDoc and an explanation why we need them? What is the purpose of `allEffectModifier`. Is it a general modifier? How do I know as user what the effect of `allEffectModifier = 3` is supposed to be?

Reply:
In general, I hesitate to put logic into components.... it often feels natural to put it there (like in this case), and I'm not sure we have a good concept for moving them somewhere else 🤔 That's probably a good discussion for Discord to drag some more people in...

Reply:
Yeah I don't have javaDocs because I haven't yet decided if I'm gonna use the allEffectModifer and was gonna remove the attribute itself as I'm not using this anywhere in this PR.. was just waiting in case some inspiration should strike..😅.. 
Removing it now though................